### PR TITLE
Validate installation source in linuxrc settings and bootloader arguments

### PR DESCRIPTION
**Pull request by mloviska**
- Related ticket: [[functional][y] gnome_http doesn't use http mirror for the installation](https://progress.opensuse.org/issues/44462)
- Verification runs:
  * [sle-15-SP1-Installer-DVD-x86_64-Build128.1-repo_inst@64bit](http://eris.suse.cz/tests/8921#step/welcome/9)
  * [sle-15-SP1-Installer-DVD-x86_64-Build128.1-gnome_smb@64bit](http://eris.suse.cz/tests/8922#step/welcome/9)
  * [sle-15-SP1-Installer-DVD-x86_64-Build128.1-gnome_http@64bit](http://eris.suse.cz/tests/8923#step/welcome/9)

Additional openQA variables in *gnome_http* test suite:
* INSTALL_SOURCE=http 
* NETBOOT=1

### CODE REVIEW

File: tests/installation/welcome.pm [hunk 1]
```diff
@@ -111,6 +111,20 @@ sub run {
         $self->accept_license;
     }
 
+    # Verify install arguments passed by bootloader
+    # Linuxrc writes its settings in /etc/install.inf
+    if (get_var('NETBOOT') && !check_var('BACKEND', 'ipmi')) {
+        wait_screen_change { send_key 'ctrl-alt-shift-x' };
+        my $method     = uc get_required_var('INSTALL_SOURCE');
+        my $mirror_src = get_var("MIRROR_$method");
+        die "Install source mismatch in boot parameters!\n"
```
Comment (ldevulder) [inline tests/installation/welcome.pm]: Maybe it's easier to read with `die` and `if` on the same line, no?

Comment (mloviska) [reply tests/installation/welcome.pm]: Thanks for feedback! I have got the impression that the line was bit long. If you find it hard to read, I am happy to change it.

Comment (rwx788) [reply tests/installation/welcome.pm]: I agree with the point that line become to long, but with two lines it's more confusing. To my taste I would use standard `if` syntax here with curly brackets.

File: tests/installation/welcome.pm [hunk 2]
```diff
@@ -111,6 +111,20 @@ sub run {
...
+          if (script_run 'grep -o --color=always install=' . $mirror_src . ' /proc/cmdline');
```
Comment (rwx788) [inline tests/installation/welcome.pm]: Let's add comparison to `0` exit code, as `script_run` is confusing as returns bash false which we interpret as perl true value. WDYT?

Comment (mloviska) [reply tests/installation/welcome.pm]: Updated, thanks for the point! Just wondering if I should add ```$rc=undef``` between the conditions. WDYT ?

Comment (rwx788) [reply tests/installation/welcome.pm]: Not really, as script_run will override previous value unconditionally. Btw, with your current code you could use one liner for the `die`, as you use the `$rc` variable ;)

Comment (mloviska) [reply tests/installation/welcome.pm]: Ah, I could realize that :D

**Comment by michelmno** [comment]
it seems this PR made bad side effect: https://openqa.opensuse.org/tests/827158#step/welcome/5 when INSTALL_SOURCE not defined.

**Comment by DimStar77** [comment]
*GRRRR!!!*

https://openqa.opensuse.org/tests/827463#step/welcome/4

If you introduce new mandatory variables, it would be REALLY awesome to define them in the test that need them.

**Comment by mloviska** [comment]
Hello all, thanks for feedback! INSTALL_SOURCE variable has not been introduced by me, I am just using it as we are used to in OSD. We have redundant ways how to deal with network installations which does not ease our lives at all. Basically we can specify remote installation source using *SUSEMIRROR* or *MIRROR_{protocol}* or *REPO_0*. Looking at migration test, I found out that it gets even more completed, because *SUSEMIRROR* is used there as boolean variable.

It would be nice if can come to some agreement how to deal with network sources within openQA tests, so we can avoid similar issues. WDYT ?

Issue should not appear anymore https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/6506

**Pull request merged**